src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -56,7 +56,8 @@ String Range::toString() const
 
 
 /// Example: for `Hello\_World% ...` string it returns `Hello_World`, and for `%test%` returns an empty string.
-String extractFixedPrefixFromLikePattern(const String & like_pattern)
+/// If perfect_prefix_match == true, only consider pattern in the format `prefix%_`
```

Comment:
To be honest, the explanation for perfect_prefix_match is not clear enough (it just gives an example but not the rule)
Let's rewrite the whole thing.

```
/// Returns the prefix of like_pattern before the first wildcard, e.g. 'Hello\_World% ...' --> 'Hello\_World', '_test%' --> ''
/// We call a prefix "perfect" if the first wildcard is only followed by other wildcards but not literals, e.g. 'test%_%' has perfect prefix 'test', 'test%_x' has no perfect prefix.